packages/block-library/src/quote/index.js

```diff
@@ -30,17 +30,23 @@ export const settings = {
 	edit,
 	save,
 	merge( attributes, { value, citation } ) {
+		// Quote citations cannot be merged. Pick the second one unless it's
+		// empty.
+		if ( ! citation ) {
+			citation = attributes.citation;
+		}
+
 		if ( ! value || value === '<p></p>' ) {
 			return {
 				...attributes,
-				citation: attributes.citation + citation,
+				citation,
 			};
 		}
 
 		return {
 			...attributes,
 			value: attributes.value + value,
```

Comment:
Should we append the first citation between the two values here?

Reply:
It feels a bit strange to me. :) If you split a quote into two quotes and merge them, you end up with a cite as quote content? In that case I would expect them to get merged in the original form. But maybe that's ok, then it requires a few more backspaces to restore. :)

Reply:
Yes, I'm not sure either, I guess I "just" wanted to avoid losing that text but it probably doesn't make a lot of sense.

Reply:
I agree with trying to avoid data loss. If the merge wasn't good, the user could always revert though. Something else we could do is ask the user which citation to use in case there are two. Kind of like a merge conflict in git. :) This pattern might also be useful to other blocks.